modules/OTAR_CRISPR.py

```diff
@@ -0,0 +1,161 @@
+
+                # Split data files to list:
+                dataFiles=study_df.dataFile.str.replace(' ', '').str.split('|')
+            )
+        )
+
+        logging.info(f'Number of studies processed: {len(self.study_df.studyId.unique())}')
+
+    def generate_evidence(self, data_folder: str) -> None:
+        """Processing the study dataframe"""
+
+        # Looping through the studies and generating evidence:
+        # Reading all data files and filter for significant hits:
+        hits = (
+            self.study_df[['studyId', 'dataFiles', 'dataFileType', 'filterColumn', 'threshold', 'projectId']]
+            .explode('dataFiles')
+            .assign(
+                dataFile=lambda df: df.apply(lambda x: f'{data_folder}/{x["projectId"]}/{x["dataFiles"]}', axis=1)
+            )
+            # TODO: parsing the data files should be file type dependent:
+            .apply(self.parse_MAGeCK_file, axis=1)
+        )
+
+        # Concatenate all hits into one single dataframe:
+        hits_df = (
+            pd.concat(hits.to_list())
+            .reset_index(drop=True)
+        )
+
+        # Merging:
+        self.merged_dataset = (
+            self.study_df
+            .assign(statisticalTestTail=lambda df: df.filterColumn.map(FILTER_COLUMN_MAP))
+            .merge(hits_df, on='studyId', how='inner')
+            .explode('diseaseFromSourceMappedId')
+            .assign(
+                datasourceId='ot_crispr',
+                datatypeId='ot_partner'
+            )
+            [[
+                'targetFromSourceId', 'diseaseFromSourceMappedId',
+                'projectId', 'studyId', 'studyOverview', 'contrast', 'crisprScreenLibrary',
+                'cellType', 'cellLineBackground', 'geneticBackground',
+                'statisticalTestTail', 'resourceScore',
+                'datasourceId', 'datatypeId'
+            ]]
+        )
+
+    @staticmethod
+    def parse_MAGeCK_file(row: pd.Series) -> pd.DataFrame:
+        """This function returns a pandas dataframe with the datafile and with properly named columns"""
+
+        datafile = row['dataFile']
+        filterColumn = row['filterColumn']
+        threshold = float(row['threshold'])
+        studyId = row['studyId']
+
+        # Read data, filter and rename columns:
+        mageck_df = (
+            pd.read_csv(datafile, sep='\t')
+            .rename(columns={filterColumn: 'resourceScore', 'id': 'targetFromSourceId'})
+            .loc[lambda df: df.resourceScore <= threshold]
+            [['targetFromSourceId', 'resourceScore']]
+            .assign(studyId=studyId)
+        )
+        logging.info(f'Number of genes reach threshold: {len(mageck_df)}')
+        return mageck_df
+
+    def write_evidence(self, output_file: str) -> None:
+        """Write the merged evidence to file"""
+
+        json_list = [json.dumps(row.dropna().to_dict()) for _, row in self.merged_dataset.iterrows()]
```

Comment:
to_json() supports exporting to JSON lines too. Do you think the dataframe could also be exported like this?

```
self.merged_dataset.dropna().to_json(output_file, compression='gzip', orient='records', lines=True)
```

Reply:
No, unfortunately that is not good. It has unwanted side effects:

```python
df = pd.DataFrame([
    {'a': 2, 'b': 24},
    {'a': 4, 'b': None},
    {'a': 12, 'b': 100}
])

(
    df
    .dropna().to_json('test.json.gz', compression='gzip', orient='records', lines=True)

)
```

Gives this:
```
{"a":2,"b":24.0}
{"a":12,"b":100.0}
```